flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1598,7 +1598,14 @@ class ExprLowering {
     // function return value.
     assert(call.getNumResults() == 1 &&
            "Expected exactly one result in FUNCTION call");
-    return call.getResult(0);
+    // TODO: should this convert be a raw cast instead ? In "real" implicit,
+    // case, we would not know the function type and would not do a type aware
+    // conversion but just pretend the result is the type we want (~ raw cast).
```

Comment:
I think you mean bitcast here.

But, yes, it's entirely possible to write bad Fortran code that the caller and callee are in total disagreement about the calling convention, arguments, or results, but the information relevant to each is internally consistent.

Bitcasts currently have to roundtrip through memory store.

Reply:
Thanks, yes, that's what I meant. If we want to support this, I will update the PR.